Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -607,66 +606,116 @@ - (void)returnBookWithIdentifier:(NSString *)identifier
     [[NYPLADEPT sharedInstance] returnLoan:fulfillmentId
                                     userID:[[NYPLUserAccount sharedAccount] userID]
                                   deviceID:[[NYPLUserAccount sharedAccount] deviceID]
-                                completion:^(BOOL success, __unused NSError *error) {
-                                  if(!success) {
+                                completion:^(BOOL success, NSError *error) {
+                                  if(success) {
+                                    [self revokeLoanAndRemoveBook:book state:state];
+                                  } else {
                                     NYPLLOG(@"Failed to return loan via NYPLAdept.");
+                                    [self presentAlertForError:error
+                                                   orErrorDict:nil
+                                                 returningBook:book];
                                   }
                                 }];
+  } else {
+    // if we have no fulfillment ID or we don't require authentication, we just
+    // need to let the Circulation Manager server know and remove the book
+    [self revokeLoanAndRemoveBook:book state:state];
   }
+#else
+  [self revokeLoanAndRemoveBook:book state:state];
 #endif
+}
 
+- (void)revokeLoanAndRemoveBook:(NYPLBook *)book state:(NYPLBookState)state
+{
+  BOOL didDownloadBook = (state == NYPLBookStateDownloadSuccessful
+                          || state == NYPLBookStateUsed);
+  NSString *const identifier = book.identifier;
+
+  // maybe this is the case when book has no DRM so there's no loan to revoke?
```

Comment:
The main time you won't see a revoke link is when the library doesn't authenticate its users. In that case, yes, there is no server-side loan to revoke.